src/java/org/apache/cassandra/db/guardrails/Guardrails.java

```diff
@@ -199,6 +201,20 @@ public final class Guardrails implements GuardrailsMBean
                    state -> CONFIG_PROVIDER.getOrCreate(state).getCompactTablesEnabled(),
                    "Creation of new COMPACT STORAGE tables");
 
+    /**
+     * Guardrail disabling the creation of a new table when {@link Option#DEFAULT_TIME_TO_LIVE} is set to 0
+     * and compaction strategy is {@link TimeWindowCompactionStrategy} or its subclass. If this guardrail does not fail
+     * and such configuration combination is found, it will always warn.
+     */
+    public static final EnableFlag zeroDefaultTTLOnTWCSDisallowed =
+    new EnableFlag("zero_default_ttl_on_twcs_disallowed",
+                   "It is suspicious to use default_time_to_live set to 0 with such compaction strategy. " +
+                   "Please keep in mind that data will not start to automatically expire after they are older " +
+                   "than a respective compaction window unit of a certain size. ",
+                   state -> CONFIG_PROVIDER.getOrCreate(state).getZeroDefaultTtlOnTimeWindowCompactionStrategyWarned(),
+                   state -> CONFIG_PROVIDER.getOrCreate(state).getZeroDefaultTtlOnTimeWindowCompactionStrategyDisallowed(),
+                   "0 default_time_to_live on a table with " + TimeWindowCompactionStrategy.class.getSimpleName() + " compaction strategy");
+
```

Comment:
I understand that what we are doing in this separate PR is replacing the 
 `_enabled` property by a `_disallowed` property with the opposite meaning. I think we can achieve that without adding a `ensureEnabled` by just reverting the property here:
```java
public static final EnableFlag zeroDefaultTTLOnTWCSDisallowed =
new EnableFlag("zero_default_ttl_on_twcs",
               "It is suspicious to use default_time_to_live set to 0 with such compaction strategy. " +
               "Please keep in mind that data will not start to automatically expire after they are older " +
               "than a respective compaction window unit of a certain size. ",
               state -> CONFIG_PROVIDER.getOrCreate(state).getZeroDefaultTtlOnTimeWindowCompactionStrategyWarned(),
               state -> !CONFIG_PROVIDER.getOrCreate(state).getZeroDefaultTtlOnTimeWindowCompactionStrategyDisallowed(),
               "0 default_time_to_live on a table with " + TimeWindowCompactionStrategy.class.getSimpleName() + " compaction strategy");
```
That would leave `EnableFlag` unchanged. It would also avoid passing `zero_default_ttl_on_twcs_disallowed` as the `enabled` argument of `EnableFlag` constructor, which feels wrong.